src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -815,18 +797,13 @@ private long getOffset() {
 
       // This api should throw exception if backoff failed.
       // fileNamesCopy after this call is emptied (clears the input list)
-      if (ingestionDeliveryGuarantee
-          == SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee.EXACTLY_ONCE) {
-        ingestionService.ingestFilesWithClientInfo(fileNamesCopy, this.clientSequencer.get());
-        String offsetToken = ingestionService.getClientStatus().getOffsetToken();
-        // Update server side offset
-        if (offsetToken == null) {
-          this.offsetPersistedInSnowflake.set(-1);
-        } else {
-          this.offsetPersistedInSnowflake.set(Long.parseLong(offsetToken));
-        }
+      ingestionService.ingestFilesWithClientInfo(fileNamesCopy, this.clientSequencer.get());
+      String offsetToken = ingestionService.getClientStatus().getOffsetToken();
+      // Update server side offset
+      if (offsetToken == null) {
+        this.offsetPersistedInSnowflake.set(-1);
       } else {
-        ingestionService.ingestFiles(fileNamesCopy);
```

Comment:
removed this else because delivery is always exactly once, so it should not fall into this else statement